lib/modules/manager/custom/types.ts

```diff
@@ -0,0 +1,17 @@
+import type { RegexManagerConfig } from './regex/types';
+
+export interface CustomExtractConfig extends Partial<RegexManagerConfig> {}
+
+export type CustomManagerName = 'regex';
+
+export interface CustomManager extends Partial<RegexManagerConfig> {
+  customType: CustomManagerName;
+  fileMatch: string[];
+}
+
+// NOTE:
+// the two interfaces might seem similar but they have different usage...
+// CustomManager interface consists of all fields that a custom manager can have
+// whereas
+// CustomExtractConfig consists of the fields that custom managers need when performing extraction
```

Comment:
```suggestion
// the two interfaces might seem similar but they are different like ManagerConfig and ExtractConfig
```
Does this seem clearer than the explanation?